mabl snippets/GetDateComponents.js

```diff
@@ -0,0 +1,184 @@
+/**
+Returns an object containing values assosiated todays date.
+IE compatiable: true
+
+To access these options, save the output of this snippet to a variable (I'll call 
+it "date" for the example) and then use this syntax: 
+    {{@<variable name>.<option>}}
+like this:
+    {{@date.monthNumeric}}
+
+OPTIONS:
+- dayNumeric
+- dayTwoDigit
+- monthNumeric
+- monthTwoDigit
+- monthNarrow
+- monthShort
+- monthLong
+- yearNumeric
+- yearTwoDigit
+- hour12TwoDigit
+- hour24TwoDigit
+- hour12Numeric
+- hour24Numeric
+- dayPeriod
+- minute
+- second
+ */
+
+function mablJavaScriptStep(mablInputs, callback) {
+  var today = new Date();
+  // Today
+  callback(getDateComponentsFor(today));
+
+  // Tomorrow
+  // callback(getDateComponentsFor(today.addDays(1)));
+
+  // Yesterday
+  // callback(getDateComponentsFor(today.addDays(-1)));
+
+  // Two weeks from now
+  // callback(getDateComponentsFor(today.addDays(14)));
+
+  // Two weeks ago
+  // callback(getDateComponentsFor(today.addDays(-14)));
+}
+
+//Adds a method to all "Date" objects in this scope
+//@param {integer} days - The number of days to add to the Date
+//@return {Date} - The new Date
+Date.prototype.addDays = function (days) {
+  var date = new Date(this.valueOf());
+  date.setDate(date.getDate() + days);
+  return date;
+};
+
+// Formats the date based off of the provided options
+// @param {object {string: string}} option - The options for how to format the date string
+// @return {string} - Formated date matching options
+// Formatting the date components:
+// Formatting Options: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/DateTimeFormat/DateTimeFormat
+Date.prototype.format = function (option) {
+  var date = new Date(this.valueOf());
+  return date.toLocaleDateString("en-US", option);
```

Comment:
I wonder if "en-US" could be made the default unless an argument is provided like in the other methods below.

Reply:
I think it can be, but for now, I just wanna get this out there 😁